packages/projection-typeorm/test/operators/storeNftMetadata.test.ts

```diff
@@ -0,0 +1,427 @@
+          );
+          const userTokenRollForwardEvt = createStubRollForwardEvent(
+            { blockBody: [txWithUserToken], blockHeader: createStubBlockHeader(Cardano.BlockNo(2)) },
+            networkInfo
+          );
+          const userTokenRollBackwardEvt = createRollBackwardEventFor(
+            userTokenRollForwardEvt,
+            referenceTokenBlockHeader
+          );
+
+          expect(await nftMetadataRepo.count()).toBe(0);
+          expect(await assetRepo.count()).toBe(0);
+
+          // Projecting reference token first should create NftMetadataEntity that is not associated with any AssetEntity.nftMetadata
+          await firstValueFrom(createStubProjectionSource([referenceTokenEvt]).pipe(applyOperators()));
+          expect(await nftMetadataRepo.count()).toBe(1);
+          expect(await assetRepo.count()).toBe(1);
+
+          // Minting user token should associate it with already existing NftMetadata
+          await firstValueFrom(createStubProjectionSource([userTokenRollForwardEvt]).pipe(applyOperators()));
+          expect(await nftMetadataRepo.count()).toBe(1);
+          expect(await assetRepo.count()).toBe(2);
+          const [nftMetadata] = await nftMetadataRepo.find({ relations: { userTokenAsset: true } });
+          expect(typeof nftMetadata.userTokenAsset?.id).toBe('string');
+          const userTokenAsset = await assetRepo.findOne({
+            relations: { nftMetadata: true },
+            where: { id: nftMetadata.userTokenAsset?.id }
+          });
+          expect(userTokenAsset?.nftMetadata?.id).toEqual(nftMetadata.id);
+
+          // Rolling back user token mint event should userToken to null
+          await firstValueFrom(createStubProjectionSource([userTokenRollBackwardEvt]).pipe(applyOperators()));
+          expect(await nftMetadataRepo.count()).toBe(1);
+          expect(await assetRepo.count()).toBe(1);
+          const nftMetadataAfterRollback = await nftMetadataRepo.findOne({
+            relations: { userTokenAsset: true },
+            where: { id: nftMetadata?.id }
+          });
+          expect(nftMetadataAfterRollback).toBeDefined();
+          expect(nftMetadataAfterRollback?.userTokenAsset?.id).toBeUndefined();
+        });
+      });
+
+      describe('user token minted first', () => {
+        it.todo('associates user nft with NftMetadata, sets nftMetadata to null when reference token is rolled back');
+      });
+    });
+  });
+
+  it('can store and retrieve files', async () => {
+    const events = await lastValueFrom(project$(withInlineDatumEvents).pipe(toArray()));
+    expect(events.length).toBeGreaterThan(1);
+    expect(await nftMetadataRepo.count()).toBeGreaterThan(0);
+    const nftAssets = await nftMetadataRepo.find({ select: ['files'] });
+    expect(nftAssets.length).toBeGreaterThan(0);
+    expect(nftAssets[0].files).toMatchObject([
+      { mediaType: 'image/gif', name: 'Shrine #2142', src: 'ipfs://QmVTzRmbmxQS3NkPcqkxjSLR4Q2oBdMbF1qtAqhJXET752' },
```

Comment:
Let's make this less brittle, e.g.
```suggestion
    const file = nftAssets.find(asset => asset.files?.length > 0)!.files![0];
    expect(typeof file.mediaType).toBe('string')
    ...
```